backend/src/impl/internal_models/system_model.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+import re
+from typing import Any
+
+from explainaboard_web.impl.db_utils.db_utils import DBUtils
+from explainaboard_web.impl.utils import abort_with_error_message, unbinarize_bson
+from explainaboard_web.models.system import System
+from explainaboard_web.models.system_info import SystemInfo
+
+
+class SystemModel(System):
```

Comment:
Do we need to use inheritance here? It seems easier to call the file `system_utils.py` and have stand-alone functions without inheritance, such as:

```
def system_from_dict(cls, dikt: dict) -> System:
```
or
```
def deserialize_system(cls, dikt: dict) -> System:
```

Reply:
Thanks for the comment! 

I remember we were using inheritance for this a while back but, at that time, I also mixed in another base class that implements general DB operations. We decided to switch to pure static methods at some point. I agree that mixing another class that handles db operations is probably too complex to maintain but I think having a class that inherits from `System` is useful here. 

`/internal_models` is intended to be the abstraction layer we discussed in #379 . I think this makes it clear that `System` is just an interface and `SytemModel` is the implementation. And we should always use `SystemModel` instead of `System` in the codebase. Besides that, it has two benefits:

1. we can store internal states in `SystemModel`.
   - use case 1: If we want an in-memory cache for `system_info`, we can easily add that as a property of the instance.
   - use case 2: If the user edits the properties of `SystemModel`, we can keep an internal state that flags the instance to be different from the one stored in the DB. And we can implement a `commit()` method that automatically commits the updates to the DB depending on the state.
 
2. (IMO) I think methods are easier to use than functions. IDEs provide better IntelliSense for class methods than standalone functions.
    - It is clear that `SystemModel.from_dict()` is an alternative constructor of `SystemModel` but it is not immediately obvious that `system_from_dict() -> System` is a constructor for `System`.
    - Similarly, it is clear that `systemModel.get_system_info()` retrieves the `system_info` for `systemModel` because it is a method that is mounted to the instance. But given `system`, it may not be obvious to the developer that there is a `SystemUtils.get_system_info(system: System)` that can be used to retrieve `system_info` for the instance.

Let me know what you think!

Reply:
OK, sounds good!